server/services/mapping_issues_service.py

```diff
@@ -45,3 +53,219 @@ def delete_mapping_issue_category(category_id: int):
     def get_all_mapping_issue_categories(include_archived):
         """ Get all mapping issue categories"""
         return MappingIssueCategory.get_all_categories(include_archived)
+
+
+class MappingIssueExporter:
+
+    def get_mapping_issues(self, project_id: int, detailedView: str, zerosRows: str):
+        """
+        Returns a csv string of all mapping issues associated with the given project summed and sorted by user
+        raises: NotFound
+        """
+        detailed = False
+        if (detailedView == "true"):
+            detailed = True
+        zeros = False
+        if (zerosRows == "true"):
+            zeros = True
+        
+        #user_dict {str username : {int task (taskId) : {str issue : str issue_count}}}
```

Comment:
Why did you leave this in the code? You probably should have either:
a) Explained why the code was commented out or
b) Removed the commented out code (preferred)

Reply:
This isn't actual code. It's comments describing what the return data types are for the following functions. I wrote these and left them because, if I were to modify anything in the get_mapping_issues method in the future, I would have to dig through what I wrote, figure out what everything is, and write them out again so I would know what I was working with. I can change them to multi-line string comments so it's clear it's not code.

Reply:
These would have been better documented in `MappingIssueExporter.compile_validated_tasks_by_user`.

For example:
```diff
diff --git a/server/services/mapping_issues_service.py b/server/services/mapping_issues_service.py
index b458b550..385847a4 100644
--- a/server/services/mapping_issues_service.py
+++ b/server/services/mapping_issues_service.py
@@ -8,6 +8,7 @@ from server.models.dtos.stats_dto import ProjectContributionsDTO
 from server.services.stats_service import StatsService
 from copy import deepcopy
 import numpy as np
+import typing
 
 class MappingIssueCategoryService:
 
@@ -182,7 +183,13 @@ class MappingIssueExporter:
 
 
     @staticmethod
-    def compile_validated_tasks_by_user(project_id: int, zeros: bool):
+    def compile_validated_tasks_by_user(project_id: int, zeros: bool) -> typing.Tuple[typing.Dict[str, typing.Dict[int, typing.Dict[str, str]]], typing.Set[any], int, typing.Dict[int, any]]:
+        """
+        :returns: user_dict: {str username: {int task (taskId) : {str issue : str issue_count}}},
+            project_users: Set[str: users on project]
+            num_validated_tasks: int
+            tasks_as_tasks_dict: {int taskId: task }
+        """
         all_project_tasks = Task.get_all_tasks(project_id)
         validated_tasks = []
         project_users = []
```

You'll want to double-check that, but it should be good for python 3.6. And you will want to replace the `any` with what it is supposed to be (`UserDTO` or `str`, or whatever).

Reply:
Note: As a bonus, if you use add the typing, most python IDEs will understand it, and give usable method possibilities.